EIPS/eip-7539.md

```diff
@@ -0,0 +1,110 @@
+---
+eip: 7539
+title: ERC Extensions
+description: Establishes Naming Conventions, Metadata, and Procedures for ERC Extensions
+author: Joey Santoro (@joeysantoro)
+discussions-to: https://ethereum-magicians.org/t/eip-7539-erc-extensions/16152
+status: Draft
+type: Meta
+created: 2023-10-18
+requires: 1
+---
+
+## Abstract
+
+This EIP Establishes a framework for ERC Extensions, or ERCs which directly extend or override the functionality of a prior ERC.
+
+This standard includes a standardized definition for an ERC Extension. It includes a naming convention for the Title EIP header field, adds an optional `extensions` EIP header field for referencing downstream ERC extensions, and extablishes procedures for specifying overridden functionality.
+
+## Motivation
+
+Ethereum application development uses a contract oriented programming paradigm which is similar to Object Oriented Programming (OOP). Because of this, many smart contract systems and high level languages have object like behavior with the possibility for overriden functionality. 
+
+ERC Standards also behave in a similar way, with logical extensions flowing from the necessity to standardize specific use cases on top of higher level standards. (e.g. [ERC-4626](./eip-4626.md) extends [ERC-20](./eip-20.md)).
+
+There should be a framework for handling these extension cases both retroactively and in the future to improve ERC accessibility.
+
```

Comment:
Your motivation should probably explain why the ad hoc conventions of today are inadequate. Essentially this section should additionally answer the question: why do we need to formalize all this instead of a note in the title and an entry in `requires`?

Reply:
There are two layers of formalization in this EIP, the definitions of the properties of an extension and the actual requirements for what constitutes an extension and how to handle them. 

I'll add a clear motivation for the actual requirements piece, and I'm open to stripping out the property formalization if you see that as getting in the way of the meat of the EIP

Reply:
Added as a suggestion above